src/figuresClasses.ts

```diff
@@ -1,11 +1,94 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  readonly shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
 
-export class Triangle implements Figure {}
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('The lenth of side is less than 0');
+    }
+
+    if (
+      this.c >= this.a + this.b ||
+      this.b >= this.c + this.a ||
+      this.a >= this.c + this.b
+    ) {
+      throw new Error(
+        `The longest side of a triangle is greater than or equal to a sum of two others`,
+      );
```

Comment:
[CODE STYLE] - If you throw an error, try to provide a detailed description in the error message. The message could be more descriptive, for example: 'The sum of any two sides of a triangle must be greater than the third side'.